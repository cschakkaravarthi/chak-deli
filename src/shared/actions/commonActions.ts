import {
  ToggleGlobalToastAction,
  TOGGLE_GLOBAL_TOAST,
  CURRENT_USER_NAME,
  SetCurrentUsername,
  SET_REJECT_MODAL_ID_ISSHOW,
  SET_REJECT_MODAL_TEXTAREA,
  SetRejectModalIdIsShow,
  SetRejectModalTextarea,
  CurrentUser
} from '../types/commonTypes';
import { ThunkAction, ThunkDispatch } from 'redux-thunk';
import { AnyAction } from 'redux';
import RejectModal from './../models/RejectModal';

export const setCurrentUsername = (currentUsername: CurrentUser): SetCurrentUsername => ({
  currentUsername,
  type: CURRENT_USER_NAME
});

const toastTimeOut = 3000;

export const toggleGlobalToast = (
  displayToast: boolean,
  toastMessage: string,
  toastError: boolean
): ToggleGlobalToastAction => ({
  displayToast,
  toastMessage,
  toastError,
  type: TOGGLE_GLOBAL_TOAST
});

export const triggerToast = (
  toastMessage: string,
  toastError = false
): ThunkAction<void, {}, {}, AnyAction> => {
  return (dispatch: ThunkDispatch<{}, {}, AnyAction>): void => {
    dispatch(toggleGlobalToast(true, toastMessage, toastError));
    // Hide toast automatically
    setTimeout(() => {
      dispatch(toggleGlobalToast(false, '', toastError));
    }, toastTimeOut);
  };
};

export const setRejectModalIdIsShow = (
  idAndIsShow: RejectModal
): SetRejectModalIdIsShow => ({
  idAndIsShow,
  type: SET_REJECT_MODAL_ID_ISSHOW
});

export const setRejectModalTextarea = (
  textarea: string
): SetRejectModalTextarea => ({
  textarea,
  type: SET_REJECT_MODAL_TEXTAREA
});

export const setReject = (
  isShow: boolean,
  sysId: string
): ThunkAction<void, {}, {}, AnyAction> => {
  return (dispatch: ThunkDispatch<{}, {}, AnyAction>): void => {
    dispatch(setRejectModalIdIsShow({ isShow: isShow, sysId: sysId }));
  };
};

export const setTextarea = (
  textarea: string
): ThunkAction<void, {}, {}, AnyAction> => {
  return (dispatch: ThunkDispatch<{}, {}, AnyAction>): void => {
    dispatch(setRejectModalTextarea(textarea));
  };
};
