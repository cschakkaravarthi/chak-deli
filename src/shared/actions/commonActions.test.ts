import { toggleGlobalToast } from './commonActions';

describe('commonActions', () => {
  it('creates a toggleGlobalToast action', () => {
    expect(toggleGlobalToast(true, 'Successfull operation!', false)).toMatchSnapshot();
  });
});
