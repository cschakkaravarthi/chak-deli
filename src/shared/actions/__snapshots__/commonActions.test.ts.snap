// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commonActions creates a toggleGlobalToast action 1`] = `
Object {
  "displayToast": true,
  "toastError": false,
  "toastMessage": "Successfull operation!",
  "type": "TOGGLE_GLOBAL_TOAST",
}
`;
